### PR TITLE
`Development`: Add server test for exercise average rating computation

### PR DESCRIPTION
**Pull request by Olimbia23**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] *This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [ ] *Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] *I added `@PreAuthorize` and checked the course groups for all new REST Calls (security).
- [x] I implemented the changes with a good performance and prevented too many database calls.
- [x] I documented the Java code using JavaDoc style.
#### Client
- [ ] *I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [ ] *I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [ ] *I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [ ] *I documented the TypeScript code using JSDoc style.
- [ ] *I added multiple screenshots/screencasts of my UI changes.
- [ ] *I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] *I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] *I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

[ ] * - not needed

### Motivation and Context

Exercise average rating and total rating was introduced in #4273 PR.
However, server integration test was missing for ensuring the correctness of the newly added functionality.
This PR solves this by adding a new server test.

I also applied the cosmetic suggestions of @Strohgelaender from the PR #4401

### Description

Added a new integration test that creates three submissions with three assessments and three corresponding ratings of 3, 4 and 5.
Then we assert to verify that the average rating and total number of ratings are computed for the given exercise.

#### Code Review
- [x] Review 1
- [x] Review 2

#### Test Coverage
AssessmentDashboardService: 97%

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java [hunk 1]
```diff
@@ -704,14 +708,25 @@ private void getCourseForDashboardWithStats(boolean isInstructor) throws Excepti
...
+        for (int i = 0; i < ratings.length; i++) {
+            var submission = database.createSubmissionForTextExercise(exercise, database.getUserByLogin("student" + (i + 1)), "text");
+            var assessment = database.addResultToSubmission(submission, AssessmentType.MANUAL, null, 0.0, true).getLatestResult();
+            database.addRatingToResult(assessment, ratings[i]);
+        }
+
+        var responseCourse = request.get("/api/courses/" + testCourse.getId() + "/for-assessment-dashboard", HttpStatus.OK, Course.class);
+        var responseExercise = responseCourse.getExercises().stream().filter(ex -> ex.getExerciseType() == ExerciseType.TEXT).findFirst().get();
+
+        // There is no average rating for exercises with no assessments and therefore no ratings
```
Comment (lschlesinger) [inline src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java:726]: is that a copy paste error? the comment does not fit to the code or am I wrong?

**Review by lschlesinger — APPROVED** [review]
Tests pass and look good!

**Review by ole-ve — APPROVED** [review]
Changes look good to me. 👍

**Review by lschlesinger — APPROVED** [review]
re-approve

**Review by Malyuk-A — APPROVED** [review]
Code looks good to me. 👍

**Review by imstreetspirit — APPROVED** [review]
it looks good to me.

**Review by ole-ve — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]
thanks for addressing my code review 👍

**Pull request merged**